src/Tribe/Traits/With_Db_Lock.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * Provides methods to acquire and release a database (SQL) lock.
+ *
+ * The MySQL functions used by this trait are `GET_LOCK`, `IS_FREE_LOCK` and `RELEASE_LOCK`.
+ * The functions are part of MySQL 5.6 and in line with WordPress minimum requirement of MySQL version (5.6).
+ *
+ * @see     https://dev.mysql.com/doc/refman/5.6/en/locking-functions.html#function_get-lock
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Traits
+ */
+
+namespace Tribe\Traits;
+
+/**
+ * Trait With_Db_Lock
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Traits
+ */
+trait With_Db_Lock {
+
+	/**
+	 * Acquires a db lock.
+	 *
+	 * To ensure back-compatibility with MySQL 5.6, the lock will hash the lock key using SHA1.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $lock_key The name of the db lock key to acquire.
+	 *
+	 * @return bool Whether the lock acquisition was successful or not.
+	 */
+	protected function acquire_db_lock( $lock_key ) {
+		global $wpdb;
+
+		/**
+		 * Filters the timeout, in seconds, of the database lock acquisition attempts.
+		 *
+		 * @since TBD
+		 *
+		 * @param int    $timeout The timeout, in seconds, of the lock acquisition attempt.
+		 * @param static $this    The object that's trying to acquire the lock by means of the trait.
+		 */
+		$timeout = apply_filters( 'tribe_db_lock_timeout', 3, $this );
```

Comment:
I would add the `$lock_key` as part of the filter. 

```suggestion
		$timeout = apply_filters( 'tribe_db_lock_timeout', 3, $lock_key, $this );
```